modules/lightning_features/lightning_media/src/MediaHelper.php

```diff
@@ -37,17 +37,26 @@ public function __construct(EntityTypeManagerInterface $entity_type_manager) {
    *   (optional) Whether to filter the bundles by create access for the current
    *   user. Defaults to FALSE.
    *
+   * @param array $target_bundles
+   *   (optional) An array of target bundles to retrieve source field extensions
+   *   for. Defaults to empty array.
+   *
    * @return string[]
    *   The file extensions accepted by all available bundles.
    */
-  public function getFileExtensions($check_access = FALSE) {
+  public function getFileExtensions($check_access = FALSE, array $target_bundles = []) {
     $extensions = '';
 
     // Lightning Media overrides the media_bundle storage handler with a special
     // one that adds an optional second parameter to loadMultiple().
-    $bundles = $this->entityTypeManager
-      ->getStorage('media_bundle')
-      ->loadMultiple(NULL, $check_access);
+    $storage = $this->entityTypeManager
+      ->getStorage('media_bundle');
+    if (!empty($target_bundles)) {
+      $bundles = $storage->loadByProperties(['id' => $target_bundles], $check_access);
```

Comment:
@phenaproxima Just realizing this as I seemed to have overlooked it previously .... The reason `loadMultiple()` cannot be used is due to the actual method's implementation to begin with:

```
  public function loadMultiple(array $ids = NULL, $check_access = FALSE) {
    if ($check_access) {
      $ids = array_filter(
        $this->getQuery()->execute(),
        [$this->accessHandler, 'createAccess']
      );
    }
    return parent::loadMultiple($ids);
  }
```

You'll notice here that the `$ids` are being overwritten from the "get all" query that is happening in `$ids = array_filter($this->getQuery()->execute() ...`

This would be the reason in which I implemented the `loadByProperties` to mimic / workaround this issue to begin with.

Thoughts?

I'd assume that this actually would result in unexpected behaviour for any Lightning based project that is using the storage handler for interacting with the media bundles...